### PR TITLE
Spruce up demos

### PR DESCRIPTION
**Pull request by applmak**
This fixes a bunch of minor little nits that were annoying me about the demo modules.

### CODE REVIEW

File: demo_modules/gears/gears_server.js [hunk 1]
```diff
@@ -111,7 +111,7 @@ export function load(debug, state, wallGeometry) {
           newZ = 1-newZ;
           // 2.5) Pick a random new radius (which might generate a random new
           // pitch), but that's okay.
-          newRadius = random.integer(35, 1000);
+          newRadius = random.integer(50, 1000);
```
Comment (jmegq) [inline demo_modules/gears/gears_server.js]: duplicated magic numbers? in my codebase?
make some consts, dude.

File: lib/math/polygon2d.js [hunk 2]
```diff
@@ -135,7 +135,7 @@ export class Polygon {
     // Step 2: Check the u values of our intersections, if any are ≈0, then we
     // hit a vertex, which means that we'll double-count our crossings.
     const numberOfCrossings = intersectionResults
-        .reduce((agg, r) => agg + (r && r.u > 0.0001 ? 1 : 0), 0);
+        .reduce((agg, r) => agg + (r && r.u > 0.00001 ? 1 : 0), 0);
```
Comment (jmegq) [inline lib/math/polygon2d.js]: comment this since you had to change it.

File: lib/shared_state.js [hunk 3]
```diff
@@ -36,26 +36,41 @@ export class SharedState {
     // Owner of the state: either a client socket id or 'server'.
     this.owner_ = owner;
   }
+  earliest() {
+    return this.store_[0];
```
Comment (jmegq) [inline lib/shared_state.js]: why no guard as with latest()?

File: demo_modules/gears/gears_server.js (hunk 1 above)
Comment (applmak) [reply demo_modules/gears/gears_server.js]: Done.

File: lib/math/polygon2d.js (hunk 2 above)
Comment (applmak) [reply lib/math/polygon2d.js]: Done.

File: lib/shared_state.js (hunk 3 above)
Comment (applmak) [reply lib/shared_state.js]: Because it will return undefined if it's not set anyway.

**Pull request merged**